### ISSUE TITLE
Multiple issues with `Client.availableDaysAndHoursUpUntil`

### ISSUE THREAD

**Issue by mikeger** [corona-warn-app/cwa-app-ios#714]
## Avoid duplicates
* [x] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [x] Bug is not already reported in another issue

## Describe the bug

1. The argument of the method `availableDaysAndHoursUpUntil(today:)` in https://github.com/corona-warn-app/cwa-app-ios/blob/next/src/xcode/ENA/ENA/Source/Client/Client%2BConvenience.swift#L52 is not used, since the portion of the code that is using it is commented. 

2. Portion of the code is commented so not fetching the hours.

3. `dispatch_group` is useless since it's only entered once.

4. Parameter name (`today`) explicitly suggests that only today's date can be passed in.

5. `Date` type shall be used to represent the date.

## Expected behaviour

See comments above.

## Steps to reproduce the issue

See https://github.com/corona-warn-app/cwa-app-ios/blob/next/src/xcode/ENA/ENA/Source/Client/Client%2BConvenience.swift#L52 

## Additional context

I acknowledge that you had to release the initial version of the app in rush, but it would be also great to make the code clean :)

**Comment by ChristianKienle** [comment]
@mikeger thank you very much for pointing this out. It is nice to see you acknowledging the timeline. In fact this was a last minute change we had to make due to a combination of different factors.

#715 should address most of your findings. As I wrote in the PR, @SebastianWild agreed to write a few more unit tests to increase the coverage in exactly that area.

You will find a lot of `String`s instead of `Date`s in the code since to a certain degree the code handles dates as ids. The API outputs those dates as strings and we store those dates as strings. Converting those strings back into dates just to convert them back to strings was not that meaningful that that time. However, we are constantly re-thinking exactly that decision.

**Comment by mikeger** [comment]
Thanks @ChristianKienle, #715 is nicely fixing it.

**Comment by mikeger** [comment]
You can link the PR to this issue. Feel free to close it once the PR is merged.
